### PR TITLE
fix: validate correctly URI Template params

### PR DESCRIPTION
**Pull request by w-vi**
Fixes https://github.com/apiaryio/drafter/issues/402

### CODE REVIEW

File: test/test-ResourceParser.cc [hunk 1]
```diff
@@ -1009,3 +1009,32 @@ TEST_CASE("Relation identifiers should be unique for a resource", "[resource]")
     REQUIRE(resource.node.actions[0].relation.str == "create");
     REQUIRE(resource.node.actions[1].relation.str == "create");
 }
+
+TEST_CASE("Detect invalid reference to URI Tempalte parameters", "[resource]")
+{
+    mdp::ByteBuffer source = \
+    "## Orders [/orders]\n\n"\
```
Comment (w-vi) [inline test/test-ResourceParser.cc]: True, I'll check

Comment (w-vi) [inline test/test-ResourceParser.cc]: So this part is a different issue, it doesn't even bother to check the parameters at all. So in this case you don't get a warning at all.

File: test/test-ResourceParser.cc [hunk 2]
```diff
@@ -1009,3 +1009,32 @@ TEST_CASE("Relation identifiers should be unique for a resource", "[resource]")
     REQUIRE(resource.node.actions[0].relation.str == "create");
     REQUIRE(resource.node.actions[1].relation.str == "create");
 }
+
+TEST_CASE("Detect invalid reference to URI Tempalte parameters", "[resource]")
```
Comment (kylef) [inline test/test-ResourceParser.cc]: Typo: Tempalte

File: test/test-ResourceParser.cc (hunk 1 above)
Comment (w-vi) [inline test/test-ResourceParser.cc]: Different PR though as it is a different issue.

Comment (pksunkara) [inline test/test-ResourceParser.cc]: Yes.

**Pull request merged**